### PR TITLE
Seed 1.0.0 prerelease

### PR DESCRIPTION
**Pull request by gisjedi**
- #89: Correct typo in spec for inputs.files.mediaType -> mediaTypes 
- #92: Added spec support and examples for partial usage
- #93: Removed outputs.files.count and replaced with outputs.files.multiple
- Various rewording and cleanup (processing system -> executor)

### CODE REVIEW

File: spec/sections/standard.adoc [hunk 1]
```diff
@@ -463,7 +472,7 @@ The following table defines the `ouputs.files` object members.
 |`name`
 |Required
 2+|MUST be a string of only alphabetic, dash or underscore characters (defined by the regex† `^[a-zA-Z_-]+$`)
-indicating the key the processing system will place the file name captured for downstream processing.
+indicating the key the executor† will place the file name captured for downstream processing.
```
Comment (gisjedi) [inline spec/sections/standard.adoc]: Self-reviewing:  This sentence sounds terrible. Suggest something better.

File: spec/sections/standard.adoc [hunk 2]
```diff
@@ -362,22 +363,28 @@ default value MUST be treated as false. If true, the `command` placeholder will
 containing all files. If false or omitted, the `command` placeholder will be replaced with an absolute path to a single
 file.
 
+|`partial`
+|Optional
+2+|MUST be a boolean indicating whether input file is required in whole or in part. This allows an executor† to make
+intelligent choices when providing very large files to a job. This should only be set to true if the job is expected to
+use less than half of very large input files. If omitted, the default value MUST be treated as true.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: Shouldn't the default for partial be false?

File: spec/sections/standard.adoc [hunk 3]
```diff
@@ -348,12 +348,13 @@ Critical implementation details related to `multiple` member should be reference
...
+the environment variable name that will be injected by the executor† for job consumption. Refer to
 <<variable-injection>> for details on environment variable available at execution time.
 
-|`mediaType`
-|Required
-2+|MUST be an array of strings that MUST indicate the IANA Media types that the job accepts.
+|`mediaTypes`
+|Optional
+2+|MUST be an array of strings that MUST indicate the IANA Media types that the job accepts. If omitted, executor† will
+not be able to perform validation.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: Suggest changing `If omitted, executor† will not be able to perform validation.` to `An executor† MAY use any provided media types to report validation warnings to the user in the case of mismatched types.`

File: spec/sections/standard.adoc [hunk 4]
```diff
@@ -348,12 +348,13 @@ Critical implementation details related to `multiple` member should be reference
 |`name`
 |Required
 2+|MUST be a string of only alphabetic, dash or underscore characters (defined by the regex† `^[a-zA-Z_-]+$`) indicating
-the environment variable name that will be injected by the processing platform for job consumption. Refer to
+the environment variable name that will be injected by the executor† for job consumption. Refer to
 <<variable-injection>> for details on environment variable available at execution time.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: Here is my suggestion:
```
MUST be a string of only alphabetic, dash or underscore characters (defined by the regex† `^[a-zA-Z_-]+$`)
indicating the unique name to use for referring to this input. All inputs will be passed as environment
variables, with the environment variable names based upon the input names. Refer to <<variable-injection>>
for details on environment variables available at execution time.
```

File: spec/sections/standard.adoc [hunk 5]
```diff
@@ -362,22 +363,28 @@ default value MUST be treated as false. If true, the `command` placeholder will
...
 
 [cols="2,1,2,5",options="header"]
 |=====
 | Name | Requirement 2+| Value
 |`name`
 |Required
 2+|MUST be a string of only alphabetic, dash or underscore characters (defined by the regex† `^[a-zA-Z_-]+$`)
-indicating the environment variable name that will be injected by the processing platform for job consumption.
+indicating the environment variable name that will be injected by the executor† for job consumption.
 Refer to <<variable-injection>> for details on environment variable available at execution time.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: I suggest the same as above.

File: spec/sections/standard.adoc [hunk 6]
```diff
@@ -401,32 +408,34 @@ MUST be treated as true.
...
+            "required": false // <10>
         }
     ]
 }
 ----
 a|
 <1> Optional array containing elements defined by Files Elements sub-section.
 <2> Required string containing name used to inject data via environment variables.
-<3> Required array containing a list of accepted Media types.
+<3> Optional array containing a list of accepted Media types. Omission will disable executor† validation.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: Media should be lower case. I would take out `Omission will disable executor† validation.`

File: spec/sections/standard.adoc [hunk 7]
```diff
@@ -401,32 +408,34 @@ MUST be treated as true.
...
 Default is `false`.
-<5> Optional boolean indicating whether job requires this particular file. Default is `true`.
-<6> Optional array containing elements defined by JSON Elements sub-section.
-<7> Required string containing name used to inject data via environment variables.
-<8> Required string containing a valid JSON schema type for input validation.
-<9> Optional boolean indicating whether job requires this particular JSON input. Default is `true`.
+<5> Optional boolean indicating that a job consumes only a small portion of input file. Default is `false`.
+<6> Optional boolean indicating whether job requires this particular file. Default is `true`.
+<7> Optional array containing elements defined by JSON Elements sub-section.
+<8> Required string containing name used to inject data via environment variables.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: Suggest `<8> Required string containing name used to refer to this input.`

File: spec/sections/standard.adoc (hunk 1 above)
Comment (dfaller) [reply spec/sections/standard.adoc]: I would suggest a similar thing as above where you just say the name is used to refer to the output. I don't think it's necessary to talk about downstream processing.

File: spec/sections/standard.adoc [hunk 8]
```diff
@@ -494,16 +503,16 @@ The following table defines the `ouputs.json` object members.
 |`name`
 |Required
 2+|MUST be a string of only alphabetic, dash or underscore characters (defined by the regex† `^[a-zA-Z_-]+$`)
-indicating the key the processing system will place the JSON member value in for downstream use. When `key` member
-is omitted, it must be a case-sensitive match of the member key in result manifest.
+indicating the key the executor† will place the JSON member value in for downstream use. When `key` member
+is omitted, it must be a case-sensitive match of the member key in `seed.outputs.json` file.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: Same here, just say name is used to refer to the output.

File: spec/sections/standard.adoc [hunk 9]
```diff
@@ -554,18 +563,18 @@ seed.outputs.json:
...
-<3> Required string containing IANA Media type of file.
-<4> Required string containing glob expression for file capture. Processing system is expected to
+<3> Optional string containing IANA Media type of file.
+<4> Required string containing glob expression for file capture. Executor† is expected to
 capture output relative to OUTPUT_DIR.
-<5> Optional string containing either a numeric count or `*` for unbounded output. Default value
-is `1`.
-<6> Optional boolean indicating whether processing system should assume failure if output data is missing. Default value
+<5> Optional boolean indicating whether a single or multiple values are supported. The executor† is expected to
+validate that matched outputs are consistent with this value - one or at least one. Default value is false.
```
Comment (dfaller) [inline spec/sections/standard.adoc]: I would remove `The executor† is expected to validate that matched outputs are consistent with this value - one or at least one.` How an executor handles matching up inputs and outputs (like for recipes) is outside of the scope of Seed.

File: spec/sections/usage.adoc [hunk 10]
```diff
@@ -58,17 +58,22 @@ a preceding switch only when the input is available:
...
 environment variable. This location MUST be made available by the executor† so that the job is given full write
 access. Developers MUST ensure any files captured by `interface.outputs.files` member element `pattern` values, as well
 as the optional `seed.outputs.json` file, are marked with read permissions at minimum. This MAY be done with the
 following sample command: `chmod +r yourfile.txt`
 
+Special attention should also be given to the number of files matched per `interface.outputs.files` element to ensure it
+is consistent in plurality to that defined by `multiple` member. If multiple files are matched using the `pattern`
+member value, while the `multiple` member is set to false, these job SHOULD be failed by the executor†. On the other
+hand, even if multiple files are not present when `multiple` is true it SHOULD NOT force a job failure by the executor†.
+
```
Comment (dfaller) [inline spec/sections/usage.adoc]: Add 's' to job on line 71.

File: spec/sections/usage.adoc [hunk 11]
```diff
@@ -58,17 +58,22 @@ a preceding switch only when the input is available:
...
 [[extended-file-metadata, Extended File Metadata]]
 ==== Extended File Metadata
-There is often a need by the processing system to capture additional job extracted metadata on output files. The Seed
+There is often a need by the executor to capture additional job extracted metadata on output files. The Seed
```
Comment (dfaller) [inline spec/sections/usage.adoc]: The word `executor` is used here without the special symbol.

**Review by dfaller — CHANGES_REQUESTED** [review]

File: spec/sections/standard.adoc (hunk 6 above)
Comment (JohnPTobe) [reply spec/sections/standard.adoc]: Agree

File: spec/sections/standard.adoc (hunk 2 above)
Comment (JohnPTobe) [reply spec/sections/standard.adoc]: Yep. Fix typo

File: spec/sections/standard.adoc (hunk 3 above)
Comment (JohnPTobe) [reply spec/sections/standard.adoc]: Seconded

File: spec/sections/standard.adoc (hunk 4 above)
Comment (JohnPTobe) [reply spec/sections/standard.adoc]: Agreed

File: spec/sections/standard.adoc (hunk 2 above)
Comment (gisjedi) [reply spec/sections/standard.adoc]: Yikes, good catch.

**Review by dfaller — APPROVED** [review]
LGTM

**Pull request merged**